r/configure

```diff
@@ -58,191 +100,282 @@ if [ ! "`${R_HOME}/bin/R CMD config CXX17`" ]; then
+  # pkg-config will handle this for us automatically, see ARROW-6312
+  if grep raspbian /etc/os-release >/dev/null 2>&1; then
+    PKG_CFLAGS="$PKG_CFLAGS -DARROW_CXXFLAGS=-latomic"
+    PKG_LIBS="-latomic $PKG_LIBS"
+  fi
+}
+
+add_feature_flags () {
+  # Now we need to check what features it was built with and enable
+  # the corresponding feature flags in the R bindings (-DARROW_R_WITH_stuff).
+  # We do this by inspecting ArrowOptions.cmake, which the libarrow build
+  # generates.
+  ARROW_OPTS_CMAKE="$LIB_DIR/cmake/Arrow/ArrowOptions.cmake"
+  if [ ! -f "${ARROW_OPTS_CMAKE}" ]; then
+    echo "*** $ARROW_OPTS_CMAKE not found; some features will not be enabled"
+  else
+    if arrow_built_with ARROW_PARQUET; then
+      PKG_CFLAGS="$PKG_CFLAGS -DARROW_R_WITH_PARQUET"
+      PKG_LIBS="-lparquet $PKG_LIBS"
+      # NOTE: parquet is assumed to have the same -L flag as arrow
+      # so there is no need to add its location to PKG_DIRS
+    fi
+    if arrow_built_with ARROW_DATASET; then
+      PKG_CFLAGS="$PKG_CFLAGS -DARROW_R_WITH_DATASET"
+      PKG_LIBS="-larrow_dataset $PKG_LIBS"
+      # NOTE: arrow-dataset is assumed to have the same -L flag as arrow
+      # so there is no need to add its location to PKG_DIRS
+    fi
+    if arrow_built_with ARROW_ACERO; then
+      PKG_CFLAGS="$PKG_CFLAGS -DARROW_R_WITH_ACERO"
+      PKG_LIBS="-larrow_acero $PKG_LIBS"
+      # NOTE: arrow-acero is assumed to have the same -L flag as arrow
+      # so there is no need to add its location to PKG_DIRS
+    fi
+    if arrow_built_with ARROW_SUBSTRAIT; then
+      PKG_CFLAGS="$PKG_CFLAGS -DARROW_R_WITH_SUBSTRAIT"
+      PKG_LIBS="-larrow_substrait $PKG_LIBS"
+      # NOTE: arrow-substrait is assumed to have the same -L flag as arrow
+      # so there is no need to add its location to PKG_DIRS
+    fi
+    if arrow_built_with ARROW_JSON; then
+      PKG_CFLAGS="$PKG_CFLAGS -DARROW_R_WITH_JSON"
+    fi
+    if arrow_built_with ARROW_S3; then
+      PKG_CFLAGS="$PKG_CFLAGS -DARROW_R_WITH_S3"
+      PKG_LIBS="$PKG_LIBS $S3_LIBS"
+    fi
+    if arrow_built_with ARROW_GCS; then
+      PKG_CFLAGS="$PKG_CFLAGS -DARROW_R_WITH_GCS"
+      PKG_LIBS="$PKG_LIBS $GCS_LIBS"
+    fi
+  fi
+}
+
+arrow_built_with() {
+  # Function to check cmake options for features
+  grep -i 'set('"$1"' "ON")' $ARROW_OPTS_CMAKE >/dev/null 2>&1
+}
+
+##############
+# Main logic #
+##############
+
+# First, find or build libarrow
+if [ "$FORCE_BUNDLED_BUILD" = "true" ]; then
+  do_bundled_build
+elif [ "$FORCE_AUTOBREW" = "true" ]; then
+  do_autobrew
+else
+  find_arrow
+  if [ "$_LIBARROW_FOUND" = "false" ]; then
+    # If we haven't found a suitable version of libarrow, build it
+    if [ "$UNAME" = "Darwin" ] && ! echo $VERSION | grep -q "000"; then
+      # Only autobrew on release version (for testing, use FORCE_AUTOBREW above)
+      # (dev versions end in .9000, and nightly gets something like .10000xxx)
+      do_autobrew
+    else
+      do_bundled_build
+    fi
+  fi
```

Comment:
Is it overkill to call out this order for the envvars in the top-matter of this file? That `FORCE_BUNDLED_BUILD` takes precedence over `FORCE_AUTOBREW` which takes precedence over the natural order of things. 

It's "obvious" when you know what those things do in detail to a certain extent, but not everyone has that in their heads.

Reply:
IMO it's overkill, and moving this function to the top will make it more visible too. I'm pretty sure the only use of FORCE_AUTOBREW is in our CI.